### PR TITLE
Bar fixes

### PR DESCRIPTION
**Pull request by n-riesco**
From https://github.com/n-riesco/plotly.js/pull/4#commitcomment-19816254

### CODE REVIEW

**Comment by etpinard** [comment]
Thanks @n-riesco !

File: src/traces/bar/hover.js [hunk 1]
```diff
@@ -67,15 +67,15 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
...
+        pointData.xLabelVal = di.b + di.s;
 
         pointData.y0 = ya.c2p(barPos(di) - barDelta, true);
         pointData.y1 = ya.c2p(barPos(di) + barDelta, true);
         pointData.yLabelVal = di.p;
     }
     else {
         pointData.y0 = pointData.y1 = ya.c2p(di.y, true);
-        pointData.yLabelVal = di.s;
+        pointData.yLabelVal = di.b + di.s;
```
Comment (etpinard) [inline src/traces/bar/hover.js]: @n-riesco this line here is breaking hover for `barmode: 'stack'` bar charts. 

Can share some insights on why you added this?

Comment (etpinard) [inline src/traces/bar/hover.js]: Great. Thanks!

Here's the simplest case http://codepen.io/etpinard/pen/Woopae

Comment (n-riesco) [inline src/traces/bar/hover.js]: So the problem is that now the bar size in non-stacked bars is di.b + di.s, whereas in stacked bars is just di.s. We could distinguish both cases just by checking whether the trace defines base. I'll submit a PR.

**Pull request merged**